app/models/check_flag_job.rb

```diff
@@ -0,0 +1,51 @@
+class CheckFlagJob < Struct.new(:flag, :token)
+
+  def perform
+    flag.access_token = token
+    flag.resolved?
+  end
+
+  def success(job)
+    flag.job_id = nil
+    flag.save
+  end
+
+  def error(job, exception)
+    @exception = exception
+  end
+
+  def failure(job)
+    flag.job_id = nil
+    flag.status = "failed"
```

Comment:
i think it's best not to update job_id or status for the CheckFlagJob. if a check job failed, it's best to just leave its status as 'submitted' and only reflect the status/job id of the submission job. you can probably simply remove the success() and failure() methods here and stick with the default ones.